File: src/main/java/core/scm/specialforms/LetSyntax.java
package core.scm.specialforms;

import core.environment.IEnvironment;
import core.evaluator.IEvaluator;
import core.scm.ISCMClass;

import java.util.List;

// TODO
public enum LetSyntax implements ISpecialForm, ISCMClass {
  LET_SYNTAX;

  private static final String syntax = "let-syntax";

  @Override
  public Object eval(List<Object> expression, IEnvironment env, IEvaluator evaluator) {
    throw new UnsupportedOperationException("NOT IMPLEMENTED YET!");
  }

  @Override
  public String toString() {
    return syntax;
  }
}

File: src/main/java/core/scm/specialforms/LetSyntax.kt
package core.scm.specialforms

import core.environment.Environment
import core.evaluator.Evaluator

// TODO
object LetSyntax : ISpecialForm {

    override fun eval(form: List<Any?>, env: Environment, evaluator: Evaluator): Any? {
        throw UnsupportedOperationException("NOT IMPLEMENTED YET!")
    }

    override fun toString() = "let-syntax"
}
